tests/networkpolicy_test.go

```diff
@@ -223,17 +291,73 @@ func assertPingFail(fromVmi, toVmi *v1.VirtualMachineInstance) {
+	policy := &networkv1.NetworkPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      name,
+		},
+		Spec: networkv1.NetworkPolicySpec{
+			PodSelector: labelSelector,
+			Ingress:     ingress,
+		},
+	}
+
+	virtClient, err := kubecli.GetKubevirtClient()
+	ExpectWithOffset(1, err).ToNot(HaveOccurred())
+
+	By(fmt.Sprintf("Create networkpolicy %s/%s", policy.Namespace, policy.Name))
+	policy, err = virtClient.NetworkingV1().NetworkPolicies(policy.Namespace).Create(policy)
+	ExpectWithOffset(1, err).ToNot(HaveOccurred(), fmt.Sprintf("should succeed creating network policy %s/%s", policy.Namespace, policy.Name))
+	return policy
+}
+
+func waitForNetworkPolicyDeletion(policy *networkv1.NetworkPolicy) {
 	if policy == nil {
 		return
 	}
 
 	virtClient, err := kubecli.GetKubevirtClient()
 	ExpectWithOffset(1, err).ToNot(HaveOccurred())
 
-	ExpectWithOffset(1, virtClient.NetworkingV1().NetworkPolicies(policy.Namespace).Delete(policy.Name, &v13.DeleteOptions{})).To(Succeed())
+	ExpectWithOffset(1, virtClient.NetworkingV1().NetworkPolicies(policy.Namespace).Delete(policy.Name, &metav1.DeleteOptions{})).To(Succeed())
 	EventuallyWithOffset(1, func() error {
-		_, err := virtClient.NetworkingV1().NetworkPolicies(policy.Namespace).Get(policy.Name, v13.GetOptions{})
+		_, err := virtClient.NetworkingV1().NetworkPolicies(policy.Namespace).Get(policy.Name, metav1.GetOptions{})
 		return err
 	}, 10*time.Second, time.Second).Should(SatisfyAll(HaveOccurred(), WithTransform(errors.IsNotFound, BeTrue())))
 }
+
+func checkHTTPPingFn(fromVmi, toVmi *v1.VirtualMachineInstance, port int) func() error {
+	return func() error {
+		const curlCheckCmd = "curl --head %s --connect-timeout 5\n"
+		for _, ip := range toVmi.Status.Interfaces[0].IPs {
+			url := fmt.Sprintf("http://%s", net.JoinHostPort(ip, strconv.Itoa(port)))
+			curlCheck := fmt.Sprintf(curlCheckCmd, url)
+			err := tests.VmiConsoleRunCommand(fromVmi, curlCheck, 10*time.Second)
+			if err != nil {
+				return fmt.Errorf("failed HTTP ping from vmi(%s/%s) to url(%s) at vmi(%s/%s): %v", fromVmi.Namespace, fromVmi.Name, url, toVmi.Namespace, toVmi.Name, err)
+			}
+		}
+		return nil
+	}
+}
+
+func assertHTTPPingSucceed(fromVmi, toVmi *v1.VirtualMachineInstance, port int) {
+	ConsistentlyWithOffset(1, checkHTTPPingFn(fromVmi, toVmi, port), 10*time.Second, time.Second).Should(Succeed())
+}
+
+func assertHTTPPingFailed(vmiFrom, vmiTo *v1.VirtualMachineInstance, port int) {
+	EventuallyWithOffset(1, checkHTTPPingFn(vmiFrom, vmiTo, port), 10*time.Second, time.Second).ShouldNot(Succeed())
+	ConsistentlyWithOffset(1, checkHTTPPingFn(vmiFrom, vmiTo, port), 10*time.Second, time.Second).ShouldNot(Succeed())
+}
+
+func assertVMIHasIPs(virtClient kubecli.KubevirtClient, vmi *v1.VirtualMachineInstance) *v1.VirtualMachineInstance {
+
+	EventuallyWithOffset(1, func() []string {
```

Comment:
i am against adding here Eventually
the interfaces must exists , and the ips, as you used WaitVMReady

this is a way to mask problems quietly imo

Reply:
Let's just remove the assert is not needed IPs are there, I moved here since IPs were being checked at assertPing functions.

Reply:
sorry i think we must make sure we dont iter on empty list and check nothing,
if you want, just put it next to the operation 
(it will not be perfect, but better than what this PR suggests at the moment, and back to master status)